integration_tests/src/janus.rs

```diff
@@ -218,3 +258,79 @@ impl<'a> Drop for Janus<'a> {
         }
     }
 }
+
+#[derive(Debug, Deserialize)]
+struct InteropFetchBatchIdsResponse {
+    status: String,
+    error: Option<String>,
+    batch_ids: Option<Vec<String>>,
+}
+
+struct JanusContainerBatchFetch {
+    http_client: reqwest::Client,
+    fetch_batch_ids_url: Url,
+}
+
+#[async_trait]
+impl BatchDiscovery for JanusContainerBatchFetch {
+    async fn get_batch_ids(&self, task_id: &TaskId) -> anyhow::Result<Vec<BatchId>> {
+        let task_id_encoded = base64::encode_config(&task_id.get_encoded(), URL_SAFE_NO_PAD);
+        let response = self
+            .http_client
+            .post(self.fetch_batch_ids_url.clone())
+            .json(&json!({
+                "task_id": task_id_encoded,
+            }))
+            .send()
+            .await?
+            .error_for_status()?
+            .json::<InteropFetchBatchIdsResponse>()
+            .await?;
+        if &response.status != "success" {
+            if let Some(error) = &response.error {
+                return Err(anyhow!(
+                    "Error from /internal/test/fetch_batch_ids: {error}"
+                ));
```

Comment:
Using the `Display` impl on `error` to construct an `anyhow::Error` is lossy. I think [`anyhow::Error::new`](https://docs.rs/anyhow/latest/anyhow/struct.Error.html#method.new) would wrap `error` while preserving its structure.

Reply:
This variable is a `String` out of a `Deserialize`-enabled struct, so we aren't losing anything. The error was necessarily flattened into a string in another process, to be passed over via JSON.